ddtrace/contrib/django/middleware.py

```diff
@@ -134,7 +134,11 @@ def _set_auth_tags(span, request):
         return span
 
     if hasattr(user, 'is_authenticated'):
-        span.set_tag('django.user.is_authenticated', user.is_authenticated())
+        try:
+            is_authenticated = user.is_authenticated()
```

Comment:
Here I think we have a couple of options:
* use a kind of wrapper in a `compat.py` module inside `contrib/django` that retrieves properly the field:
```python
def is_authenticated(user):
    if callable(user.is_authenticated):
        return user.is_authenticated()
    return user.is_authenticated
```
* ask forgiveness and not the permission (that is idiomatic for Python and is what you're doing), but in this case we hit the exception most of the time (so it's not an exception anymore). Any newer version of Django (>=2.0) will have only the boolean and so we'll always hit the `except` block, introducing the overhead to handle the exception for any request.

Even if I really like your approach, would be great making `user.is_authenticated ` the default case and not the exception since it will be future-proof.

Does it make sense? What do you think?

Reply:
Yea, the exception probably isn't great for performance. I've updated the PR with a version check and a wrapper in compat.py.